tests/foreman/cli/test_contentview.py

```diff
@@ -353,6 +353,84 @@ def test_positive_delete_version_by_id(self, module_org, module_product, module_
         new_cv = module_target_sat.cli.ContentView.info({'id': new_cv['id']})
         assert len(new_cv['versions']) == 0
 
+    @pytest.mark.tier2
+    def test_negative_publish_during_repo_sync(
+        self,
+        module_target_sat,
+        module_org,
+        module_product,
+    ):
+        """Attempt to publish a new version of the content-view,
+        while an associated repository is being synced.
+
+        :id: 94b53947-9b1d-414d-8319-f9a24e9170a6
+
+        :BZ: 1957144
+
+        :setup: A newly created repository and content-view.
+
+        :steps:
+            1. Add a larger repository to the content-view.
+            2. Perform asynchronous repository sync.
+            3. Attempt to publish a version of the content-view, while repo sync ongoing.
+
+        :expectedresults:
+            1. User cannot publish during repository sync.
+            2. CLIReturnCodeError raised, publish task failed for expected reason,
+                repo sync task_id found in humanized error, content-view versions unchanged.
+        """
+        custom_repo = module_target_sat.cli_factory.make_repository(
+            {
+                'content-type': 'yum',
+                'product-id': module_product.id,
+                'url': settings.repos.rhel8_os.appstream,
+            }
+        )
+        content_view = module_target_sat.cli_factory.make_content_view(
+            {'organization-id': module_org.id}
+        )
+        # Associate repo to created CV
+        module_target_sat.cli.ContentView.add_repository(
+            {
+                'id': content_view['id'],
+                'organization-id': module_org.id,
+                'repository-id': custom_repo['id'],
+            }
+        )
+        # read updated CV info
+        content_view = module_target_sat.cli.ContentView.info({'id': content_view['id']})
+        existing_versions = content_view['versions']
+        # perform async repository sync
+        repo_task = module_target_sat.cli.Repository.synchronize(
+            {
+                'id': custom_repo['id'],
+                'async': True,
+            }
+        )
+        repo_task_id = repo_task.split()[-1].rstrip('.')
+        result = None
+        # attempt to publish a new version of the content view
+        with pytest.raises(CLIReturnCodeError) as err:
+            result = module_target_sat.cli.ContentView.publish({'id': content_view['id']})
+        assert 'Could not publish' in err.value.stderr
+        assert repo_task_id in err.value.stderr
+        assert result is None
```

Comment:
Non-blocking, just out of curiosity - why this? Is it possible to get out of `with pytest.raises` with the proper error (asserted above) and still have some result from CLI at the same time?

Reply:
The `err` fetched is of type `ExceptionInfo`, to get the actual exception, I call `ExceptionInfo.value`
``` 
<ExceptionInfo CLIReturnCodeError(status=70, stderr='Could not publish the content view:\n  Pending tasks detected in repositories of...-168-90.rhos-01.prod.psi.rdu2.redhat.com/foreman_tasks/tasks/069b219c-5a18-4d36-9ea8-97f0dec7ea45 before publishing.\n' tblen=4> 
```

Reply:
We could also assert both are in `str(err)`, but for thoroughness I wanted to check the exception object itself, 
instead of just a stringified representation.

```
(Pdb) type(err)
class <ExceptionInfo>

(Pdb) dir(err)
[ ... , _stringify_exception', '_striptext', '_traceback', 'errisinstance', 'exconly', 'fill_unfilled', 'for_later', 'from_current', 'from_exc_info', 'from_exception', 'getrepr', 'group_contains', 'match', 'tb', 'traceback', 'type', 'typename', 'value']

(Pdb) err.value
CLIReturnCodeError(status=70, stderr...
```

Reply:
No no, I'm sorry, I haven't described my question properly. The `err` assertions are clear, I just meant the very last line - `assert result is None` - I believe the `result` would be always `None` when an exception is raised in `sat.cli.whatever_fails()` and caught in `with pytest.raises`. Unless I'm missing something, which is quite possible too.

Reply:
oh gotcha right! I did this assertion just to ensure nothing was returned by the CLI command.
Assuming it didn't reach the exception, should fail before this assertion.
But, would it be possible for the exception to be raised, but some CLI output to still be returned in error? I did this assertion just to be thorough lol

Reply:
I'm guessing from `module_target_sat.cli.ContentView.publish()` the code runs [execute](https://github.com/SatelliteQE/robottelo/blob/3f11c65dfd5ddb880c526e04a97f1430f17ce2d0/robottelo/cli/contentview.py#L114), which just runs the command [here](https://github.com/SatelliteQE/robottelo/blob/3f11c65dfd5ddb880c526e04a97f1430f17ce2d0/robottelo/cli/base.py#L203) and uses the `_handle_response` bellow which, according to [this](https://github.com/SatelliteQE/robottelo/blob/3f11c65dfd5ddb880c526e04a97f1430f17ce2d0/robottelo/cli/base.py#L51-L63), throws one or the other error for non-zero return code or returns the stdout otherwise.
So it IMHO should not happen - you get error or stdout, not both. And if you don't get the error, `pytest.raises` fails your test.

Reply:
removed this assertion for command output. 
The prior API test did not check for any result besides exception, 
so let's keep consistent between the two 👍